### ISSUE TITLE
smithy-go version incompatibility

### ISSUE THREAD

**Issue by asterikx** [aws/aws-sdk-go-v2#936]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though the [API reference](https://docs.aws.amazon.com/sdk-for-go/v2/api/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-go) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-go-v2/issues) and didn't find any solution
  
**Describe the bug**
Running `go get -u ./...` adds `github.com/awslabs/smithy-go v0.4.0 // indirect` to `go.mod` and fails:

```
# github.com/aws/aws-sdk-go-v2/service/sts
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/sts@v0.29.0/api_op_AssumeRole.go:327:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/sts@v0.29.0/api_op_AssumeRoleWithSAML.go:279:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/sts@v0.29.0/api_op_AssumeRoleWithWebIdentity.go:305:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/sts@v0.29.0/api_op_DecodeAuthorizationMessage.go:88:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/sts@v0.29.0/api_op_GetAccessKeyInfo.go:77:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/sts@v0.29.0/api_op_GetCallerIdentity.go:72:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/sts@v0.29.0/api_op_GetFederationToken.go:250:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/sts@v0.29.0/api_op_GetSessionToken.go:130:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/sts@v0.29.0/endpoints.go:101:2: too many arguments to return
        have ("github.com/awslabs/smithy-go/middleware".SerializeMiddleware, error)
        want (error)
# github.com/aws/aws-sdk-go-v2/service/rdsdata
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/rdsdata@v0.29.0/api_op_BatchExecuteStatement.go:101:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/rdsdata@v0.29.0/api_op_BeginTransaction.go:77:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/rdsdata@v0.29.0/api_op_CommitTransaction.go:69:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/rdsdata@v0.29.0/api_op_ExecuteSql.go:81:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/rdsdata@v0.29.0/api_op_ExecuteStatement.go:116:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/rdsdata@v0.29.0/api_op_RollbackTransaction.go:71:2: undefined: "github.com/awslabs/smithy-go/transport/http".AddContentLengthMiddleware
../../../go/pkg/mod/github.com/aws/aws-sdk-go-v2/service/rdsdata@v0.29.0/endpoints.go:101:2: too many arguments to return
        have ("github.com/awslabs/smithy-go/middleware".SerializeMiddleware, error)
        want (error)
```

**Version of AWS SDK for Go?**
v0.29.0

**Version of Go (`go version`)?**
go1.15.5 darwin/amd64

**Additional context**
PR #935 might resolve this issue.  It would be great if this PR could be released soon.

**Comment by jasdel** [comment]
Thanks for reporting this issue @asterikx . We've merged in #935 to update the sdk side of these changes. A tagged release of the sdk will be release earlier next week for these changes.

**Comment by asterikx** [comment]
I can confirm that the issue has been resolved in `v0.30.0`.
